labs-ios-starter/MapScreen/MapViewController.swift

```diff
@@ -0,0 +1,183 @@
+    
+    // MARK: - Properties
+    
+    var city: City?
+    let categories = CategoryList()
+    var span: MKCoordinateSpan = MKCoordinateSpan(latitudeDelta: 0.1, longitudeDelta: 0.1)
+    var mapItems: [POIAnnotation] = []
+    
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setUpView()
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func backButtonPressed(_ sender: Any) {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @IBAction func zoomChangedValue(_ sender: UISlider) {
+        span.latitudeDelta = 0.1 - CLLocationDegrees(sender.value)
+        span.longitudeDelta = 0.1 - CLLocationDegrees(sender.value)
+        mapView.setRegion(MKCoordinateRegion(center: mapView.region.center, span: span), animated: true)
+    }
+    
+    @IBAction func zoomInPressed(_ sender: UIButton) {
+        if span.latitudeDelta >= 0.02 {
+            span.latitudeDelta -= 0.01
+            span.longitudeDelta -= 0.01
+            zoomSlider.setValue(zoomSlider.value + 0.01, animated: true)
+        } else {
+            span.latitudeDelta = 0.01
+            span.longitudeDelta = 0.01
+            zoomSlider.setValue(0.09, animated: true)
+        }
+        mapView.setRegion(MKCoordinateRegion(center: mapView.region.center, span: span), animated: true)
+    }
+    
+    @IBAction func zoomOutPressed(_ sender: UIButton) {
+        if span.latitudeDelta <= 0.09 {
+            span.latitudeDelta += 0.01
+            span.longitudeDelta += 0.01
+            zoomSlider.setValue(zoomSlider.value - 0.01, animated: true)
+        } else {
+            span.latitudeDelta = 0.1
+            span.longitudeDelta = 0.1
+            zoomSlider.setValue(0, animated: true)
+        }
+        mapView.setRegion(MKCoordinateRegion(center: mapView.region.center, span: span), animated: true)
+    }
+    
+    @IBAction func poiChosen() {
+        let category = categories.cats[pickerView.selectedRow(inComponent: 0)]
+        requestPointsOfInterest(category: category)
+    }
+    
+    // MARK: - Private Functions
+    
+    private func setUpView() {
+        guard let city = city else { return }
+        cityLabel.text = city.cityName
+        pickerView.delegate = self
+        pickerView.dataSource = self
+        tableView.delegate = self
+        tableView.dataSource = self
+        mapView.delegate = self
+        let annotations = mapView.annotations
+        mapView.removeAnnotations(annotations)
+        let coordinate: CLLocationCoordinate2D = CLLocationCoordinate2DMake(city.latitude, city.longitude)
+        let region = MKCoordinateRegion(center: coordinate, span: span)
+        mapView.setRegion(region, animated: true)
+        mapView.register(MKMarkerAnnotationView.self, forAnnotationViewWithReuseIdentifier: ReuseIdentifier.poiAnnotation)
+    }
+    
+    private func requestPointsOfInterest(category: POICategory) {
+        if #available(iOS 14.0, *) {
```

Comment:
I would recommend you update your minimum deployment version to iOS 14 so that you don't need to use these flags.